mat/lu.go

```diff
@@ -19,11 +19,62 @@ const (
 	badLU          = "mat: invalid LU factorization"
 )
 
-// LU is a type for creating and using the LU factorization of a matrix.
+// LU is a square n×n matrix represented by its LU factorization with partial
+// pivoting.
+//
+// The factorization has the form
+//
+//	A = P * L * U
+//
+// where P is a permutation matrix, L is lower triangular with unit diagonal
+// elements, and U is upper triangular.
+//
+// Note that this matrix representation is useful for certain operations, in
+// particular for solving linear systems of equations. It is very inefficient at
+// other operations, in particular At is slow.
 type LU struct {
 	lu    *Dense
 	swaps []int
+	piv   []int
 	cond  float64
+	ok    bool // Whether A is singular
```

Comment:
This is a little confusing; the intention is that `!ok` → singular, but it could be read differently.

Reply:
Good point, thanks. I've also adjusted comments in the involved Lapack routines.